tests/integration/nm/linux_bridge_test.py

```diff
@@ -313,3 +313,85 @@ def test_bridge_enslave_unmanaged_interface(nm_unmanaged_dummy1):
             }
         )
         libnmstate.apply(desired_state)
+
+
+@pytest.fixture
+def nm_down_unmanaged_dummy1():
+    cmdlib.exec_cmd(
+        f"ip link add name {DUMMY1} type dummy".split(" "), check=True
+    )
+    yield
+    cmdlib.exec_cmd(
+        f"nmcli dev set {DUMMY1} managed yes".split(" "), check=True
+    )
+    time.sleep(1)  # Wait device became managed
+    libnmstate.apply(
+        {
+            Interface.KEY: [
+                {
+                    Interface.NAME: DUMMY1,
+                    Interface.STATE: InterfaceState.ABSENT,
+                }
+            ]
+        },
+        verify_change=False,
+    )
+
+
+@pytest.mark.tier1
+def test_reapply_bridge_state_does_not_managed_ports(nm_down_unmanaged_dummy1):
+    with linux_bridge(
+        BRIDGE0,
+        bridge_subtree_state={
+            LB.OPTIONS_SUBTREE: {LB.STP_SUBTREE: {LB.STP.ENABLED: False}}
+        },
+    ) as desired_state:
+        cmdlib.exec_cmd(
+            f"ip link set {DUMMY1} master {BRIDGE0}".split(), check=True
+        )
+        libnmstate.apply(desired_state)
```

Comment:
based on the usage of `sleep()` above, don't we need to have one here, too, so that NM has a chance to manage the interface?

Reply:
Here we are using Nmstate implementation which uses async. calls and event/signals. So Nmstate is waiting until all the operations finished.